### PR TITLE
Add support for Realtime RC

### PR DESCRIPTION
**Pull request by cynthiajoan**
Add support for Realtime RC, via an event that users can subscribe to for config updates.
***

https://github.com/firebase/firebase-unity-sdk/actions/runs/4877992103
And testing locally
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: remote_config/src/FirebaseRemoteConfig.cs [hunk 1]
```diff
@@ -34,11 +34,41 @@ public sealed class FirebaseRemoteConfig {
...
+    // Function for C++ to call when the config is updated.
+    private static RemoteConfigUtil.ConfigUpdateDelegate configUpdateDelegate =
+    new RemoteConfigUtil.ConfigUpdateDelegate(ConfigUpdateMethod);
+
     /// @brief App object associated with this FirebaseRemoteConfig.
     public FirebaseApp App {
       get { return firebaseApp; }
     }
 
+    private event EventHandler<ConfigUpdateEventArgs> ConfigUpdateListnerImpl;
```
Comment (AlmostMatt) [inline remote_config/src/FirebaseRemoteConfig.cs:46]: nitpick: typo Listner => Listener

Comment (a-maurice) [reply remote_config/src/FirebaseRemoteConfig.cs]: Fixed

File: remote_config/src/swig/remote_config.i [hunk 2]
```diff
@@ -49,36 +60,61 @@ struct ConfigValueInternal {
   ValueSource source;
 };
 
+// Wrapper that calls g_token_changed. Should be used with the
```
Comment (AlmostMatt) [inline remote_config/src/swig/remote_config.i:63]: calls g_config_updated

**Review by AlmostMatt — APPROVED** [review]
I've verified with Testapp that these methods work. (Added a listener, and the listener was called).

A couple nitpicks about copy-pasted comments

File: remote_config/src/swig/remote_config.i [hunk 3]
```diff
@@ -49,36 +60,61 @@ struct ConfigValueInternal {
...
+// callback logic to guarantee it is on the Unity thread.
+static void CallConfigUpdate(ConfigUpdate cu, RemoteConfigError error, const char* name) {
+  if (g_config_updated) {
+    // Should be calling FirebaseAppCheck.TokenChangedMethod
+    g_config_updated(name, &cu, static_cast<int>(error));
+  }
+}
+// Called by C# to register a RemoteConfig instance for token change updates,
```
[the remark this answers is not in the document]
Comment (a-maurice) [reply remote_config/src/swig/remote_config.i]: Done

File: remote_config/src/swig/remote_config.i (hunk 2 above)
Comment (a-maurice) [reply remote_config/src/swig/remote_config.i]: Done

**Pull request merged**